.pre-commit-config.yaml

```diff
@@ -0,0 +1,90 @@
+---
+repos:
+- repo: local
+  hooks:
+  - id: immutable-setup-py
+    name: Verify that setup.py stays immutable
+    description: >-
+      This is a sanity check that makes sure that
+      the `setup.py` file isn't changed.
+    # Using Python here because using
+    # shell test does not seem to work in CIs:
+    entry: >-
+      sh -c 'git hash-object setup.py
+      |
+      python -c raise\ SystemExit\(input\(\)\ !=\ \"f6d1010b609cbe816d3ef652eee452d09d52979f\"\)
+      '
+    pass_filenames: false
+    language: system
+    files: >-
+      ^setup\.py$
```

Comment:
I'm hearing `pip` now special-cases setuptools to support editable installs w/o `setup.py`. Somebody needs to evaluate how well that works and maybe just wipe this file.